bookkeeper-server/src/main/java/org/apache/bookkeeper/client/SyncCallbackUtils.java

```diff
@@ -83,12 +84,12 @@
         @Override
         @SuppressWarnings(value = "unchecked")
         public void createComplete(int rc, LedgerHandle lh, Object ctx) {
-            SyncCallbackUtils.finish(rc, lh, this);
+            SyncCallbackUtils.finish(rc, lh, (CompletableFuture<LedgerHandle>) this);
```

Comment:
hmm... this is a bit tricky. you are casting a CompletableFuture<T> to CompletableFuture<LedgerHandle>. did you take a look the approach I suggested?